lib/apiservers/engine/backends/push.go

```diff
@@ -0,0 +1,135 @@
+package backends
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	"github.com/vmware/vic/lib/apiservers/engine/backends/cache"
+	vicarchive "github.com/vmware/vic/lib/archive"
+	"github.com/vmware/vic/lib/imagec"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/sys"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/api/types"
+	eventtypes "github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/docker/docker/reference"
+)
+
+// PushImage initiates a push operation on the repository named localName.
+func (i *Image) PushImage(ctx context.Context, image, tag string, metaHeaders map[string][]string, authConfig *types.AuthConfig, outStream io.Writer) error {
+	// return fmt.Errorf("%s does not yet implement image.PushImage", ProductName())
+	defer trace.End(trace.Begin(fmt.Sprintf("%s:%s", image, tag)))
+
+	_, err := cache.ImageCache().Get(image)
```

Comment:
No, we need the check here.  It's our standard practice.  Also, even though imagec is part of the persona, it's a separate library.  We should not NOT do a check because we know a library code will  also do the check.  The responsibility of this check should be in the backend.